src/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -136,21 +136,7 @@ private void Initialize()
                 // machine connecting to public server).
 
                 // SMTP RFC's require ASCII only host names in the HELO/EHLO message.
-
-                // TODO: https://github.com/dotnet/corefx/issues/19605
-                //
-                //   Catching NotSupportedException here because HostName is currently PNSE
-                //   on UapAot. HostName is planned to be restored for .NetCore 2.1
-                //   so once that's done, the try-catch can be removed.
-                string clientDomainRaw;
-                try
-                {
-                    clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
-                }
-                catch (NotSupportedException)
-                {
-                    clientDomainRaw = "LocalHost";
-                }
+                string clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
```

Comment:
@caesar1995 I didn't realize this is called by the ctor. A platform policy is that synchronous I/O (or anything blocking) is prohibited on the UI thread of an UWP application (it will crash the app). 
In general, we should avoid calling I/O during construction of objects (not sure if this is the case with GetIPGlobalProperties()).

Otherwise, I'm OK with the change if it's covered by existing tests.